### PR TITLE
Move ember-decorators to devDeps

### PR DESCRIPTION
**Pull request by rwwagner90**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,6 @@
     "lint:files": "ember adde-lint --file-names"
   },
   "dependencies": {
-    "@ember-decorators/argument": "^0.8.16",
     "@ember-decorators/babel-transforms": "^0.1.1",
```
Comment (ezpuzz) [inline package.json]: what about this one?

Comment (rwwagner90) [reply package.json]: I think we need the babel transforms, since we were using classes still, but not sure.

File: addon/-private/column-tree.js [hunk 2]
```diff
@@ -508,16 +508,19 @@ export default EmberObject.extend({
...
   },
 
   destroy() {
     this.token.cancel();
     get(this, 'root').destroy();
 
-    removeObserver(this, 'columns.@each.isFixed', this.sortColumnsByFixed.bind(this));
-    removeObserver(this, 'widthConstraint', this.ensureWidthConstraint.bind(this));
+    removeObserver(this, 'columns.@each.isFixed', this._sortColumnsByFixed);
+    removeObserver(this, 'widthConstraint', this._ensureWidthConstraint);
```
Comment (mixonic) [inline addon/-private/column-tree.js]: This is a good fix, though it seems unrelated to the PR description 🤔
